tests/unit/test_ai_loop.py

```
@@ -0,0 +1,946 @@
+"""Tests for AI loop behavior across interactive modes (local, remote, auto).
+
+Covers three primary scenarios:
+- Mode 'local': CLI interactive prompts for permissions and follow-up
+- Mode 'remote': DB-polling for permission and follow-up confirmations
+- Mode 'auto': Non-interactive, permissions blocked, stop tool exits loop
+"""
+import json
+import unittest
+from dataclasses import field
```

Flake8 (7.3.0)

[error] 'dataclasses.field' imported but unused

(F401)